src/Microsoft.AspNetCore.Http.Extensions/RequestHeaders.cs

```diff
@@ -249,6 +249,18 @@ public RangeConditionHeaderValue IfRange
             }
         }
 
+        public Uri Origin
+        {
+            get
+            {
+                return Headers.Get<Uri>(HeaderNames.Origin);
+            }
+            set
+            {
+                Headers.Set(HeaderNames.Origin, value);
```

Comment:
We can't use the default Uri.ToString, it is not properly formatted for headers. See how we do it for the Location header.
https://github.com/vanillajonathan/HttpAbstractions/blob/e7814525527f4f6ea7a62beae23f1a9f57b1b038/src/Microsoft.AspNetCore.Http.Extensions/ResponseHeaders.cs#L132-L147

Reply:
Okay, I think I fixed this. I did write the getter and setter in the same way as for the Location header.